tests/disk.rs

```diff
@@ -0,0 +1,62 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+#[test]
+#[cfg(all(feature = "system", feature = "disk"))]
+fn test_disks() {
+    if sysinfo::IS_SUPPORTED_SYSTEM {
+        let s = sysinfo::System::new_all();
+        // If we don't have any physical core present, it's very likely that we're inside a VM...
+        if s.physical_core_count().unwrap_or_default() > 0 {
+            let mut disks = sysinfo::Disks::new();
+            assert!(disks.list().is_empty());
+            disks.refresh_list();
+            assert!(!disks.list().is_empty());
+        }
+    }
+}
+
+#[test]
+#[cfg(feature = "disk")]
+fn test_disks_usage() {
+    use std::io::Write;
+    use tempfile::NamedTempFile;
+
+    if !sysinfo::IS_SUPPORTED_SYSTEM {
+        return;
+    }
+
+    let mut disks = sysinfo::Disks::new_with_refreshed_list();
+
+    let mut file = NamedTempFile::new().unwrap();
```

Comment:
I wonder if this test is failing because temp files on linux are actually not written on disk. Maybe try with a temp folder within which you add a file?

Reply:
It works fine locally on my Ubuntu 22.04 machine, but I did try out your suggestion and unfortunately, still the same result. I am wondering if this is some limitation with how the Ubuntu CI runner is set up...

Reply:
Let's use the trick used above:

```rust
        // If we don't have any physical core present, it's very likely that we're inside a VM...
        if s.physical_core_count().unwrap_or_default() > 0 {
```

Like that, the test will only run locally. I'll try to figure out later on what's wrong.